components/datetime/src/scaffold/get_field.rs

```diff
@@ -205,6 +219,13 @@ impl<C: Calendar, A: AsCalendar<Calendar = C>, Z> GetField<DayOfYear> for ZonedD
     }
 }
 
+impl<C: Calendar, A: AsCalendar<Calendar = C>, Z> GetField<RataDie> for ZonedDateTime<A, Z> {
+    #[inline]
+    fn get_field(&self) -> RataDie {
+        self.date.to_rata_die()
```

Comment:
Question/Issue: is `to_rata_die` efficient in general? The fields are evaluated eagerly, and this is wasteful in the 99% case where the pattern doesn't contain this field. There's a long-standing open issue to read the fields lazily but it hasn't been a priority because all of them are currently fast to compute.